lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,10 @@
 
 def palindrome_permutation?(string)
-  raise NotImplementedError, "palindrome_permutation? not implemented"
+  return true if string.length == 0
+
+  hash = Hash.new(0)
+  string.each_char { |ele| hash[ele] += 1 }
+  
+  hash.values.one?(1) ? (return true) : (return false)
```

Comment:
This assumes the extra character would only occur  once, what if it occurs a different number of times?

You might do better to count the number of odd values.